example-app/src/main/java/com/adyen/checkout/example/ui/settings/composable/SettingsScreen.kt

```diff
@@ -0,0 +1,267 @@
+import com.adyen.checkout.example.R
+import com.adyen.checkout.example.ui.compose.UIText
+import com.adyen.checkout.example.ui.settings.model.EditSettingsData
+import com.adyen.checkout.example.ui.settings.model.SettingsCategory
+import com.adyen.checkout.example.ui.settings.model.SettingsIdentifier
+import com.adyen.checkout.example.ui.settings.model.SettingsItem
+import com.adyen.checkout.example.ui.settings.model.SettingsUIState
+import com.adyen.checkout.example.ui.settings.viewmodel.SettingsViewModel
+import com.adyen.checkout.example.ui.theme.ExampleTheme
+
+@Composable
+internal fun SettingsScreen(
+    onBackPressed: () -> Unit,
+    viewModel: SettingsViewModel = hiltViewModel(),
+) {
+    Scaffold(
+        modifier = Modifier.windowInsetsPadding(WindowInsets.ime),
+        topBar = {
+            TopAppBar(
+                title = { Text(text = stringResource(id = R.string.settings_screen_title)) },
+                navigationIcon = {
+                    IconButton(onClick = onBackPressed) {
+                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Back")
+                    }
+                },
+            )
+        },
+    ) { innerPadding ->
+        val uiState by viewModel.uiState.collectAsState()
+        SettingsScreen(
+            uiState = uiState,
+            onItemClicked = viewModel::onItemClicked,
+            onEditSettingConsumed = viewModel::onEditSettingConsumed,
+            onTextSettingChanged = viewModel::onTextSettingChanged,
+            onListSettingChanged = viewModel::onListSettingChanged,
+            onSwitchSettingChanged = viewModel::onSwitchSettingChanged,
+            modifier = Modifier.padding(innerPadding),
+        )
+    }
+}
+
+@Suppress("LongParameterList")
+@Composable
+private fun SettingsScreen(
+    uiState: SettingsUIState,
+    onItemClicked: (SettingsItem) -> Unit,
+    onEditSettingConsumed: () -> Unit,
+    onTextSettingChanged: (SettingsIdentifier, String) -> Unit,
+    onListSettingChanged: (SettingsIdentifier, EditSettingsData.SingleSelectList.Item) -> Unit,
+    onSwitchSettingChanged: (SettingsIdentifier, Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    SettingsItemsList(
+        settingsCategories = uiState.settingsCategories,
+        onItemClicked = onItemClicked,
+        onSwitchSettingChanged = onSwitchSettingChanged,
+        modifier = modifier,
+    )
+
+    if (uiState.settingToEdit != null) {
+        EditSettingDialog(
+            settingToEdit = uiState.settingToEdit,
+            onTextSettingChanged = {
+                onTextSettingChanged(uiState.settingToEdit.identifier, it)
+                onEditSettingConsumed()
+            },
+            onListSettingChanged = {
+                onListSettingChanged(uiState.settingToEdit.identifier, it)
+                onEditSettingConsumed()
+            },
+            onDismiss = {
+                onEditSettingConsumed()
+            },
```

Comment:
```suggestion
            onTextSettingChanged = {
                onTextSettingChanged(uiState.settingToEdit.identifier, it)
            },
            onListSettingChanged = {
                onListSettingChanged(uiState.settingToEdit.identifier, it)
            },
            onDismiss = {
                onEditSettingDismissed()
            },
```

What do you think of changing the `onEditSettingConsumed()` to have a more descriptive name, like `onEditSettingConsumed()` and only call it when the dialog is dismissed? Since calling `onTextSettingChanged()` or `onListSettingChanged()` means that the settings is changed and the dialog should hide right?